galsim/wfirst/wfirst_detectors.py

```diff
@@ -82,22 +82,54 @@ def applyIPC(img, edge_treatment='extend', fill_value=None):
     """
     img.applyIPC(galsim.wfirst.ipc_kernel, edge_treatment=edge_treatment, fill_value=fill_value)
 
-def allDetectorEffects(img, rng=None, exptime=None):
+def applyPersistence(img, prev_exposures):
+    """
+    Applies the persistence effect to the Image instance by adding a small fraction of the previous
+    exposures (up to 8) supplied as the 'prev_exposures' argument.
+
+    For more information about persistence, see the docstring for galsim.Image.applyPersistence.
+    Unlike that routine, this one does not need the coefficients to be specified. However, the list
+    of previous eight exposures will have to be supplied. Earlier exposures, if supplied, will be
+    ignored.
+
+    @param img               The Image to be transformed.
+    @param prev_exposures    List of up to 8 Image instances in the order of exposures, with the
+                             recent exposure being the first element.
+    """
+    if hasattr(prev_exposures,'__iter__'):
+        n_exp = min(len(prev_exposures),len(galsim.wfirst.persistence_coefficients))
+        if n_exp > len(galsim.wfirst.persistence_coefficients):
+            import warnings
+            warnings.warn("More than 8 Image instances were passed to"
+                          " galsim.wfirst.applyPersistence routine. Ignoring the earlier"
+                          " exposures")
+        img.applyPersistence(prev_exposures[:n_exp],galsim.wfirst.persistence_coefficients[:n_exp])
+    else:
+        raise TypeError("In wfirst.applyPersistence, 'prev_exposures' must be a list of Image"
+        " instances")
```

Comment:
Likewise here.  It would be cleaner to check for the error and raise an exception at the start rather than delay the code for the exception to the end in an else clause.